### PR TITLE
fix: Always create cookie store with network context (3-1-x)

### PR DESCRIPTION
**Pull request by deepak1556**
##### Description of Change

Backports https://crrev.com/c/1152083
Ref https://github.com/electron/electron/pull/15836

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] Patch information is added to appropriate `.patches.yaml`
- [x] `script/update` runs without error
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: patches/common/chromium/network_context.patch [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ #include "mojo/public/cpp/bindings/strong_binding_set.h"
+-#include "services/network/cookie_manager.h"
+ #include "services/network/public/mojom/network_service.mojom.h"
+ #include "services/network/public/mojom/udp_socket.mojom.h"
+ #include "services/network/public/mojom/url_loader_factory.mojom.h"
+@@ -28,6 +27,7 @@ class URLRequestContext;
+ }  // namespace net
+ 
+ namespace network {
++class CookieManager;
```
Comment (felixrieseberg) [inline patches/common/chromium/network_context.patch:106]: I see the trouble now 😆

**Review by felixrieseberg — APPROVED** [review]
This makes sense to me after looking at https://crrev.com/c/1152083 (once tests pass).

**Comment by felixrieseberg** [comment]
```
../../services/network/network_context.cc:323:12: error: member reference type 'network::URLRequestContextBuilderMojo *' is a pointer; did you mean to use '->'?
    builder.SetCookieAndChannelIdStores(std::move(cookie_store),
    ~~~~~~~^
           ->
```

**Review by nornagon — APPROVED** [review]
Seems reasonable to me

**Review by codebytere — APPROVED** [review]

**Comment by deepak1556** [comment]
@codebytere lets merge this, the 3-1-x libcc bump will use the merge commit and these builds are useless.

**Pull request merged**